core/templates/pages/practice-session-page/practice-session-page.component.ts

```diff
@@ -104,11 +105,13 @@ angular.module('oppia').component('practiceSessionPage', {
             'Practice Session: ' + ctrl.topicName + ' - Oppia');
         });
       };
+
       ctrl.$onInit = function() {
         ctrl.topicName = UrlService.getTopicUrlFragmentFromLearnerUrl();
         ctrl.stringifiedSubtopicIds = (
           UrlService.getSelectedSubtopicsFromUrl());
         _fetchSkillDetails();
+        PreventPageUnloadEventService.addListener();
```

Comment:
Can you prevent the page unload ONLY WHEN the user is in the middle of the practice session? Something like: if they just open the practice session, but don't interact with it or are still on the first question, then they are free to close the tab without any popup. Seems like you can pass a parameter to `addListener` though I haven't looked into this yet.